### PR TITLE
feat: load config from toml file

### PR DESCRIPTION
**Pull request by amenomori**
It is disturbing that json with comments always failing with lint. So I made this PR to help the config files become campatible of TOML files.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,32 +1,36 @@
 module github.com/Simple-Tracker/qBittorrent-ClientBlocker
 
-go 1.20
+go 1.21.0
```
Comment (Simple-Tracker) [inline go.mod:3]: To maintain compatibility, it is not recommended to upgrade the Go version. (Go 1.21 drops support for Windows 7, but there are no features that require urgent updates for now)
As an alternative, you can specify the go-toml version v2.2.2.

File: go.mod [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
+
+toolchain go1.23.0
```
Comment (Simple-Tracker) [inline go.mod:5]: Doesn't seem necessary?

File: config.toml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+checkUpdate = true
+debug = false
+interval = 6
+cleanInterval = 3_600
+banTime = 86_400
+timeout = 6
+logToFile = true
+
+clientType = ""
+clientURL = "http://qbittorrent"
```
Comment (Simple-Tracker) [inline config.toml:10]: Although it is helpful to understand, it is different from the default empty value "" in config.

File: config.toml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+blockListFile = [
+    "blockList.json",
+    # "blockList-Optional.json"
+]
+blockListURL = [
+    "https://bta.iaalai.cn/qBittorrent-ClientBlocker/blockList.json",
+    "https://cdn.jsdelivr.net/gh/Simple-Tracker/qBittorrent-ClientBlocker@dev/blockList.json",
+]
+portBlockList = []
+ipBlockList = ["ipBlockList.txt"]
```
Comment (Simple-Tracker) [inline config.toml:26]: This is a mistake in the previous default configuration, it should be ipBlockListFile.

**Review by Simple-Tracker — COMMENTED** [review]
Thanks for the PR!

This is a good change and will help users choose the configuration filetype that is more suitable for them. Although I see some minor issues.

BTW: How is the support for comments in toml?

File: config.go [hunk 5]
```diff
@@ -460,9 +461,17 @@ func LoadConfig(filename string, notExistErr bool) int {
 
 	configLastMod[filename] = tmpConfigLastMod
 
-	if err := json.Unmarshal(jsonc.ToJSON(configFile), &config); err != nil {
-		Log("LoadConfig", GetLangText("Error-ParseConfig"), true, filename, err.Error())
-		return -4
+	switch filepath.Ext(filename) {
```
Comment (Simple-Tracker) [inline config.go:464]: It is recommended to convert it to lowercase (strings.ToLower)

**Comment by amenomori** [comment]
> Thanks for the PR!
> 
> This is a good change and will help users choose the configuration filetype that is more suitable for them. Although I see some minor issues.
> 
> BTW: How is the support for comments in toml?

As for comments, please refer to [the official spec](https://toml.io/en/v1.0.0#comment).

File: config.go (hunk 5 above)
Comment (amenomori) [reply config.go:464]: Fixed.

File: config.toml (hunk 3 above)
Comment (amenomori) [reply config.toml:10]: Wrong value by mistake.

File: config.toml (hunk 4 above)
Comment (amenomori) [reply config.toml:26]: Fixed. The name in `config.json` also changed.

File: go.mod (hunk 1 above)
Comment (amenomori) [reply go.mod:3]: Specify the version to `v2.2.2`.

File: go.mod (hunk 2 above)
Comment (amenomori) [reply go.mod:5]: Introduced by go since 1.21. Removed here.

**Comment by Simple-Tracker** [comment]
Thanks! It's been merged.

**Pull request merged**